File: Medium/3133. Minimum Array End/sol.cpp
class Solution {
public:
    long long minEnd(int n, int x) {
        // long named result and set it to x as it will be out first element
        long long result = x;
        // loop from 1 to n-1 as we are already setting the first element as x
        for(int i =1; i < n; i++){
            // add 1 to result and OR it with x
            result = (result + 1) | x;
        }
        // return result
        return result;
    }
};

File: Medium/3133. Minimum Array End/sol.py
# https://leetcode.com/problems/minimum-array-end?envType=daily-question&envId=2024-11-09
#  this works but leds to time limit exceeded
# cpp solves that

class Solution:
    def minEnd(self, n: int, x: int) -> int:
        result = x

        # we need one less because the first element will be x itslef
        for i in range(n-1):
            result += 1
            # and we OR it guranteing that all the bits are set in result here
            result = result | x

        return result
    
